dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java

```diff
@@ -567,6 +569,19 @@ private Response getFileByToolVersionID(String registryId, String versionIdParam
         return Response.status(status).build();
     }
 
+    public static List<Checksum> getFileDescriptorChecksumsForTrs(final SourceFile sourceFile) {
+        List<Checksum> trsChecksums = new ArrayList<>();
+        if (sourceFile.getChecksums() != null && !sourceFile.getChecksums().isEmpty()) {
```

Comment:
If `sourceFile.getChecksums()` returns a non-empty value, this method will return an empty ArrayList, which on line 520 you will use to overwrite the existing checksums.

Reply:
My comment was wrong about line 520; you are using it with a new Dockerfile, so there won't be an existing checksum that gets overridden.